### DISCUSSION TITLE
Now that IRazorComponentApplication<T> is removed, what is the replacement to discover endpoint for Blazor SSR?

### DISCUSSION THREAD

**Discussion by dodyg** [dotnet/aspnetcore#48544] [Q&A] [answered]
https://github.com/dotnet/aspnetcore/pull/48283/files#diff-a7f3c6db39f6146badbd23e53b266014ab84a78b73e26998961b2f1ab62ce32f

My simple sample for SSR no longer works using Preview 6

https://github.com/dodyg/practical-aspnetcore/blob/net6.0/projects/.net8/RazorComponentOne/README.md

**Comment by gfoidl** [comment]
Did you see https://github.com/dotnet/aspnetcore/issues/46980#issuecomment-1460509030?
(Note: this is not my domain, so I hope to gave you the right pointer)

**Reply by dodyg** [reply]
Yeah I have read those. It doesn't have any useful specific unfortunately.

**Reply by gfoidl** [reply]
@javiercn please shed some light into this question.

**Reply by javiercn** [reply, marked as answer]
We got rid of the interface, we are working on a mechanism that is based on source generation, but the public entry point for any of this is MapRazorComponents<TRoot>() that exposes a `ComponentApplicationBuilder` instance

**Reply by javiercn** [reply]
Is there a concrete scenario you are trying to achieve?

**Reply by dodyg** [reply]
I just want the following code to work
``` csharp 

var builder = WebApplication.CreateBuilder();
builder.Services.AddRazorComponents();

var app = builder.Build();

app.MapRazorComponents<RazorComponentOne.Shared.MainLayout>();

app.Run();
```

``` razor
@page "/"

Good morning hello world

<Greetings Message="Hello World" />
```

**Reply by javiercn** [reply]
@dodyg put an App component and add a router inside, exactly how you would do in Blazor Server or Web assembly.

**Reply by dodyg** [reply]
@javiercn I added the App.razor and now the layout renders. However the body of the layout doesn't. 

![msg-1](https://github.com/dotnet/aspnetcore/assets/1761249/ca4db527-632c-4666-809a-3c8334618347)
![msg-2](https://github.com/dotnet/aspnetcore/assets/1761249/11de0bb8-1648-40fe-81a7-7b3e40a170fb)
![msg-3](https://github.com/dotnet/aspnetcore/assets/1761249/5898cbe0-7572-4ed3-a476-f1124f9b801f)

**Reply by javiercn** [reply]
MapRazorComponents should use the App component, not the main layout. You can download the nightly bits, create an instance of the new template and compare it against your code.
